dask/dataframe/io/parquet/arrow.py

```diff
@@ -185,12 +185,45 @@ def write(df, fs, fs_token, path, append=False, partition_on=None, **kwargs):
         ]
         return delayed(writes)
 
+    @staticmethod
+    def create_metadata(df, fs, path, append=False, partition_on=None,
+                        ignore_divisions=False, **kwargs):
+        fs.mkdirs(path)
+        if ignore_divisions:
+            raise NotImplementedError("`ignore_divisions` not implemented"
+                                      " for `engine='pyarrow'`")
+        object_encoding = kwargs.pop("object_encoding", "utf8")
+        if object_encoding == "infer" or (
+            isinstance(object_encoding,
+                       dict) and "infer" in object_encoding.values()
+        ):
+            raise ValueError(
+                '"infer" not allowed as object encoding, '
+                "because this required data in memory."
+            )
```

Comment:
I think that `object_encoding` is strictly a fastparquet thing.  We can probably drop it safely.